File: core-examples/src/main/java/io/vertx/example/core/verticle/worker/MainVerticle.java
package io.vertx.example.core.verticle.worker;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.DeploymentOptions;
import io.vertx.core.Launcher;
import io.vertx.core.ThreadingModel;

/**
 * An example illustrating how worker verticles can be deployed and how to interact with them.
 *
 * This example prints the name of the current thread at various locations to exhibit the event loop <-> worker
 * thread switches.
 */
public class MainVerticle extends AbstractVerticle {

  public static void main(String[] args) {
    Launcher.executeCommand("run", MainVerticle.class.getName());
  }

  @Override
  public void start() throws Exception {
    System.out.println("[Main] Running in " + Thread.currentThread().getName());
    vertx
      .deployVerticle("io.vertx.example.core.verticle.worker.WorkerVerticle",
        new DeploymentOptions().setThreadingModel(ThreadingModel.WORKER));

    vertx.eventBus().request(
      "sample.data",
      "hello vert.x").onComplete(r -> {
        System.out.println("[Main] Receiving reply ' " + r.result().body()
          + "' in " + Thread.currentThread().getName());
      }
    );
  }
}

File: core-examples/src/main/kotlin/io/vertx/example/core/verticle/worker/MainVerticle.kt
package io.vertx.example.core.verticle.worker

import io.vertx.core.DeploymentOptions
import io.vertx.kotlin.core.*

class MainVerticle : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    println("[Main] Running in ${java.lang.Thread.currentThread().getName()}")
    vertx.deployVerticle("io.vertx.example.core.verticle.worker.WorkerVerticle", DeploymentOptions(
      worker = true))

    vertx.eventBus().request<Any>("sample.data", "hello vert.x", { r ->
      println("[Main] Receiving reply ' ${r.result().body()}' in ${java.lang.Thread.currentThread().getName()}")
    })
  }
}
